packages/gatsby/src/query/__tests__/graphql-runner.ts

```diff
@@ -30,10 +31,12 @@ beforeAll(() => {
   store.dispatch({ type: `DELETE_CACHE` })
 })
 
-const reporter = require(`gatsby-cli/lib/reporter`)
 afterEach(() => {
   reporter.error.mockClear()
   reporter.warn.mockClear()
+
+  store.getState().schemaCustomization.composer.clear()
+  store.getState().schemaCustomization.types = []
```

Comment:
When using GraphqlCompose in v7 you could override  deprecatedReason with undefined and it would go away. In v9, you can't undeprecate a field by running createTypes again which to me sounds like the correct behavior. I can mimic old behaviour if we feel strongly about it.

That's why I now clear the composer and the types for each test so they run in a clean environment.